boot/assets.go

```diff
@@ -50,19 +162,60 @@ type TrustedAssetsInstallObserver struct {
 //
 // Implements gadget.ContentObserver.
 func (o *TrustedAssetsInstallObserver) Observe(op gadget.ContentOperation, affectedStruct *gadget.LaidOutStructure, root, realSource, relativeTarget string) (bool, error) {
-	// TODO:UC20:
-	// steps on write action:
-	// - copy new asset to assets cache
-	// - update modeeenv
+	if affectedStruct.Role != gadget.SystemBoot {
+		// only care about system-boot
+		return true, nil
+	}
+
+	if o.blName == "" {
+		// we have no information about the bootloader yet
+		bl, err := bootloader.ForGadget(o.gadgetDir, root, &bootloader.Options{NoSlashBoot: true})
+		if err != nil {
+			return false, fmt.Errorf("cannot find bootloader: %v", err)
+		}
+		tbl, ok := bl.(bootloader.TrustedAssetsBootloader)
+		if !ok {
+			return true, nil
+		}
+		trustedAssets, err := tbl.TrustedAssets()
+		if err != nil {
+			return false, fmt.Errorf("cannot list %q bootloader trusted assets: %v", bl.Name(), err)
+		}
+		o.trustedAssets = trustedAssets
+		o.blName = bl.Name()
```

Comment:
this can move earlier to avoid doing Find again for bootloaders without trusted assets